wasmedge_plugin_sdk/src/sdk/plugin.rs

```diff
@@ -188,11 +188,10 @@ where
 impl Placeholder for OptionString {
     fn create_placeholder() -> Box<Self> {
         let length: u32 = 128;
-        let buf = vec![0i8; length as usize].into_boxed_slice();
-        let buf_ptr = Box::into_raw(buf) as *mut i8;
+        let buf: Vec<u8> = vec![0u8; length as usize];
```

Comment:
This vec will be dropped when the function ends, but its address is being used by OptionString, which seems to be a potential memory safety issue. Are you sure there’s no problem here?

Reply:
I think you are right. How about this one.
```rust
let buf = vec![0u8; length as usize];
let buf_ptr = Box::into_raw(Box::new(buf)) as *const u8;
```

or I could just roll back to the original style.
```rust
let buf = vec![0i8; length as usize].into_boxed_slice();
let buf_ptr = Box::into_raw(buf) as *mut i8;
```

If we want to completely eliminate the use of Box (I'm not sure if it's feasible in here). However, I am currently occupied with the schedule for other projects, so for now, I'll focus on resolving the compile errors first before addressing this task later.

Reply:
I think you should roll back to the previous version. Leaking is better than accessing memory that has already been released.

Reply:
Rollback and add a drop implementation, restore ptr to box. I hope this can solve the memory leak issue.